quickstarts/develop.markdown

```diff
@@ -0,0 +1,98 @@
+Kroxylicious' composable filter chains and pluggable API mean that you can write your own filters and apply your own rules to the Kafka protocol.
+
+In this quickstart you will build a custom filter and use it to modify messages being sent to/consumed from Kafka, learn how filter configuration works, discover how to run Kroxylicious with custom filters, and find a starting point for developing your own custom filters with your own rules and logic.
+
+<br />
+
+## Getting started
+
+#### Prerequisites
+
+To start developing your own custom filters for Kroxylicious, you will need to install **both** [JDK 17](https://openjdk.org/projects/jdk/17/) and [JDK 21](https://openjdk.org/projects/jdk/21/).
+This is because the integration tests require newer language features that aren't available in JDK 17. Alternatively, you can exclude the `integrationtests` module when you build to avoid needing to install JDK 21.
+
+You'll also need to install the [Apache Maven CLI](https://maven.apache.org/index.html) and one of either [Podman](https://podman.io/docs/installation) or [Docker](https://docs.docker.com/install/) (Note that if you are using Podman, you may encounter issues with the integration tests. There are instructions [here](https://github.com/kroxylicious/kroxylicious/blob/main/DEV_GUIDE.md#running-integration-tests-on-podman) to resolve this).
+
+<br />
+
+#### Get the code
+
+The easiest way to learn how to build custom filters is with our `kroxylicious-sample` module, which contains some basic find-and-replace filters for you to experiment with.
+Begin by downloading the latest `kroxylicious-sample` sources from the [Kroxylicious repository](https://github.com/kroxylicious/kroxylicious).
+
+```shell
+git clone git@github.com:kroxylicious/kroxylicious.git
+```
+
+<br />
+
+## Build
+
+Building the sample project is easy! You can build the `kroxylicious-sample` jar either on its own or with the rest of the Kroxylicious project.
+
+To build all of Kroxylicious, including the sample:
+
+```shell
+mvn verify
+```
+
+Build with the `dist` profile for creating executable JARs:
+
+```shell
+mvn verify -Pdist -Dquick
+```
+
+<br />
+
+## Run
+
+Build both `kroxylicious-sample` and `kroxylicious-app` with the `dist` profile as above, then run the following command:
+
+```shell
+KROXYLICIOUS_CLASSPATH="kroxylicious-sample/target/*" kroxylicious-app/target/kroxylicious-app-*-bin/kroxylicious-app-*/bin/kroxylicious-start.sh --config kroxylicious-sample/sample-proxy-config.yml
```

Comment:
We should definitely replace this with a shell script. Either generated as part of the dist profile or just copied in. (obviously thats out of scope for this PR)